redis/tests/test_cluster_async.rs

```diff
@@ -897,45 +898,52 @@ mod cluster_async {
             runtime,
             async_connection: mut connection,
             ..
-        } = MockEnv::new(name, move |cmd: &[u8], port| {
-            if !started.load(atomic::Ordering::SeqCst) {
-                respond_startup_with_replica_using_config(
-                    name,
-                    cmd,
-                    Some(slots_config_vec[0].clone()),
-                )?;
-            }
-            started.store(true, atomic::Ordering::SeqCst);
+        } = MockEnv::with_client_builder(
+            ClusterClient::builder(vec![&*format!("redis://{name}")])
+                // Disable the rate limiter to refresh slots immediately on all MOVED errors.
+                .slots_refresh_rate_limit(Duration::from_secs(0), 0),
```

Comment:
The only change in this test is adding slots_refresh_rate_limit and adding missing handling for "contains_slice(cmd, b"SETNAME")", other than that it's only indentation changes